src/openpersonen/api/serializers/ingeschreven_personen.py

```diff
@@ -93,63 +94,27 @@ def add_expand_data(self, instance, representation):
                 raise ValueError(param)
-
-        for param in query_params:
-            if "." in param:
+            elif "." in param:
                 self.handle_dot_notation(param, instance, representation)
             else:
                 representation[param] = getattr(instance, param)
-                for param in representation[param]:
-                    param["url"] = self.get_links_url(
-                        instance.burgerservicenummer, param
+                for expand_field in representation[param]:
+                    expand_field["url"] = self.get_links_url(
+                        instance.burgerservicenummer, expand_field
                     )
 
     def add_links(self, instance, representation):
 
-        representation["partners_links"] = []
-        for partner in instance.partners:
-            representation["partners_links"].append(
-                {
-                    "url": self.context["request"]
-                    .build_absolute_uri()
-                    .split("?")[0]
-                    .replace("/" + instance.burgerservicenummer, "")
-                    + "/"
-                    + instance.burgerservicenummer
-                    + "/partners/"
-                    + partner["burgerservicenummer"]
-                }
-            )
-
-        representation["kinderen_links"] = []
-        for kind in instance.kinderen:
-            representation["kinderen_links"].append(
-                {
-                    "url": self.context["request"]
-                    .build_absolute_uri()
-                    .split("?")[0]
-                    .replace("/" + instance.burgerservicenummer, "")
-                    + "/"
-                    + instance.burgerservicenummer
-                    + "/kinderen/"
-                    + kind["burgerservicenummer"]
-                }
-            )
-
-        representation["ouders_links"] = []
-        for ouder in instance.ouders:
-            representation["ouders_links"].append(
-                {
-                    "url": self.context["request"]
-                    .build_absolute_uri()
-                    .split("?")[0]
-                    .replace("/" + instance.burgerservicenummer, "")
-                    + "/"
-                    + instance.burgerservicenummer
-                    + "/ouders/"
-                    + ouder["burgerservicenummer"]
-                }
-            )
+        for field in self.expand_fields:
+            representation[f"{field}_links"] = []
+            for obj in getattr(instance, field):
+                representation[f"{field}_links"].append(
+                    {
+                        "url": self.context["request"].build_absolute_uri(self.context["request"].path).replace(
+                            f'/{instance.burgerservicenummer}', '')
+                               + f"/{instance.burgerservicenummer}/{field}/{obj['burgerservicenummer']}"
```

Comment:
Can't you replace it with the correct path directly, instead of replacing it with an empty string and then appending? Or just skip the replace and append `f"/{instance.burgerservicenummer}/{field}/{obj['burgerservicenummer']}"`?
```suggestion
"url": self.context["request"].build_absolute_uri(self.context["request"].path).replace(
                            f'/{instance.burgerservicenummer}', f"/{instance.burgerservicenummer}/{field}/{obj['burgerservicenummer']}")
```

Reply:
`self.context["request"].build_absolute_uri(self.context["request"].path)` won't always have the instance's burgerservicenummer.  It depends on whether the url you call is `/api/ingeschrevenpersonen/` which does not have it while `/api/ingeschrevenpersonen/{burgerservicenummer}` does have it.  

For this reason we can't rely on the burgerservicenummer being there which is why I remove it if it is there and then always add it back.

Reply:
So ideally what I want is to always return the url `/api/ingeschrevenpersonen/` but I don't know of a way to do this when calling `/api/ingeschrevenpersonen/{burgerservicenummer}` which is why I do the replace with an empty string to remove it.

Reply:
Ahh I see, then it's good :+1: